qiskit/synthesis/multi_controlled/mcx_synthesis.py

```diff
@@ -333,10 +365,10 @@ def _linear_depth_ladder_ops(num_ladder_qubits: int) -> tuple[QuantumCircuit, li
     qc = QuantumCircuit(n)
     qreg = list(range(n))
 
+    relative_ccx_x = _n_parallel_ccx_x(1)
     # up-ladder
     for i in range(2, n - 2, 2):
-        qc.ccx(qreg[i + 1], qreg[i + 2], qreg[i])
-        qc.x(qreg[i])
+        qc.compose(relative_ccx_x, [qreg[i + 1], qreg[i + 2], qreg[i]], inplace=True)
```

Comment:
one last comment: why do you need to call here `relative_ccx_x` and can't use `rccx` and `x` directly?
similar comment for lines 380 and 383.